scaleway/resource_instance_server.go

```diff
@@ -284,16 +284,43 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 			Size: scw.Size(uint64(size.(int)) * gb),
 		}
 	}
+	serverType := getServerType(instanceAPI, req.Zone, req.CommercialType)
+	if serverType != nil {
+		// We had a local root volume if it is not already present
+		if rootVolume := req.Volumes["0"]; rootVolume == nil {
+			req.Volumes["0"] = &instance.VolumeTemplate{
+				Name:       newRandomName("vol"),
+				VolumeType: instance.VolumeVolumeTypeLSSD,
+				Size:       serverType.VolumesConstraint.MinSize,
+			}
+		}
+	}
```

Comment:
Should this be in an else of the if just above ?

Reply:
`serverType` should be defined outside anyway because of the `validateLocalVolumeSizes` that come later that requires it.